athena_cfn/athena-cfn.yml

```
@@ -0,0 +1,146 @@
+# Purpose:  Enable Athena functionality from Quilt web catalog
+#===============================================================================
+# aws cloudformation validate-template --template-body file://athena/athena_cfn.yml
+#  Scoped to ${AWS::StackName}
+
+AWSTemplateFormatVersion: "2010-09-09"
+Description: |
+  Enable Athena from Quilt - Creates Athena Workgroup, Bucket, Policy
+
+Parameters:
+  DatabaseName:
+    Default: default
+    NoEcho: false
+    Description: Name of an EXISTING Athena database to use for Quilt tables
+    Type: String
+    MinLength: 1
+    MaxLength: 16
+    AllowedPattern: '[a-z][_a-z]*'
+  DatabaseTableRegex:
+    Default: '*_quilt_*'
+    NoEcho: false
+    Description: Regexp for Quilt-created Tables
+    Type: String
+  SuffixBucket:
+    Default: output
+    NoEcho: false
+    Description: Suffix for per-workgroup Athena output bucket
+    Type: String
+    MinLength: 3
+    AllowedPattern: '[a-z][a-z0-9]*'
+  SuffixWorkGroup:
+    Default: query
+    NoEcho: false
+    Description: Suffix for Quilt-specific Athena workgroup
+    Type: String
+    MinLength: 3
+    AllowedPattern: '[a-z][a-z0-9]*'
+
+# Non-parameters (no intrinsics allowed)
+# ARNDatacatalog: !Sub 'arn:aws:glue:${AWS::Region}:${AWS::AccountId}:datacatalog/*'
+# ARNCatalog:!Sub 'arn:aws:glue:${AWS::Region}:${AWS::AccountId}:catalog'
+
+Resources:
+  QuiltAthenaOutput:
+    Type: "AWS::S3::Bucket"
+    #DeletionPolicy: Retain
+    Properties:
+      BucketName: !Join ["-", [!Ref AWS::StackName, !Ref SuffixWorkGroup, !Ref SuffixBucket]]
+
+  QuiltAthenaWorkgroup:
+    Type: AWS::Athena::WorkGroup
+    DeletionPolicy: Delete
+    Properties:
+      Name: !Join ["-", [!Ref AWS::StackName, !Ref SuffixWorkGroup]]
+      RecursiveDeleteOption: true
+      WorkGroupConfiguration:
+        PublishCloudWatchMetricsEnabled: true
+        ResultConfiguration:
+          OutputLocation: !Join ["", ["s3://" , Ref: QuiltAthenaOutput, "/"]]
```

IMHO `Sub` looks better here (and in many other places):
```suggestion
          OutputLocation: !Sub 's3://${QuiltAthenaOutput}/'
```

---

Huh.  I thought I could not reference local variables directly in a !Sub, but had to explicitly construct a map. That's why I used Join instead.

---

Cool, that works!